### PR TITLE
wait for failed jobs

### LINKED ISSUE #9285: --wait-for-jobs does not fails if job failed
Output of `helm version`: 
version.BuildInfo{Version:"v3.5.0", GitCommit:"32c22239423b3b4ba6706d450bd044baffdcf9e6", GitTreeState:"clean", GoVersion:"go1.15.6"}

Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"archive", BuildDate:"2020-11-25T13:19:56Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"9f2892aab98fe339f3bd70e3c470144299398ace", GitTreeState:"clean", BuildDate:"2020-08-13T16:04:18Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

I have this template:

```
apiVersion: batch/v1
kind: Job
metadata:
  name: backend-tests-job
  labels:
    app-name: tests
    app-service: backend-tests
spec:
  backoffLimit: 0
  template:
    metadata:
      name: backend-tests
    spec:
      restartPolicy: Never
      imagePullSecrets:
        - name: {{ .Values.imagePullSecrets }}

      containers:
      - name: tests
        image: {{.Values.image.repository}}/{{ .Values.appName }}:{{ .Values.appVersion }}
        imagePullPolicy: Always
        command: ["/usr/bin/make"]
        args: ["test", "PYTEST_ARGS=${ARGS}", "VERSION=${VERSION}"]
        env:
          - name: ARGS
            value: {{ .Values.pytestArgs | quote }}
          - name: GG_VERSION
            value: {{ .Values.version | quote }}
        resources:
            {{- toYaml .Values.resources | nindent 12 }}

```

I installs it with this command:
`helm install CHART --wait-for-jobs --wait --timeout 10m`

Sometimes job failed (because of tests, yeap :) ), does not restart (because of backoffLimit: 0) and then helm just... waits. 
I think that helm should not wait if job failed without any other restarts

### PR DESCRIPTION
**Pull request by antonio-antuan**
**What this PR does / why we need it**: closes #9285 

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [*] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/kube/wait_test.go [hunk 1]
```diff
@@ -266,6 +266,16 @@ func Test_waiter_jobReady(t *testing.T) {
...
 		},
+		{
+			name: "job is completed with backoffLimit=0",
+			args: args{job: newJob("foo", 0, 1, 1, 0)},
+			want: true,
+		},
+		{
+			name: "job is failed and it is ok",
+			args: args{job: newJob("foo", 0, 1, 0, 1)},
+			want: true,
```
Comment (jdolitsky) [inline pkg/kube/wait_test.go:277]: Hi @aCLr - can you verify if this PR solves the same issue? https://github.com/helm/helm/pull/9313

I noticed this `want: true` is the opposite as the test provided in that PR?

**Review by jdolitsky — CHANGES_REQUESTED** [review]

File: pkg/kube/wait_test.go (hunk 1 above)
Comment (aCLr) [reply pkg/kube/wait_test.go:277]: No this is not the same. My PR closes #9313 while #9285 can't do the same.
Currently we use that patch in production and everything works like a charm.

Comment (jdolitsky) [reply pkg/kube/wait_test.go:277]: @aCLr - sorry can you clarify? You use _this_ patch, or the one provided in #9285 ?

**Comment by jouve** [comment]
> This branch has conflicts that must be resolved

@aCLr can you  rebase ?

**Comment by joejulian** [comment]
@antonio-antuan If this is still relevant, could you fix the conflicts? If not, can you close it?

**Comment by antonio-antuan** [comment]
> @antonio-antuan If this is still relevant, could you fix the conflicts? If not, can you close it?

ok closed

**Pull request closed without merging**